pkg/aws/secret.go

```diff
@@ -19,50 +19,77 @@ import (
 	"fmt"
 
 	awsclient "github.com/gardener/gardener-extension-provider-aws/pkg/aws/client"
-	extensionscontroller "github.com/gardener/gardener/extensions/pkg/controller"
 
+	extensionscontroller "github.com/gardener/gardener/extensions/pkg/controller"
 	corev1 "k8s.io/api/core/v1"
+	"k8s.io/utils/pointer"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 // GetCredentialsFromSecretRef reads the secret given by the the secret reference and returns the read Credentials
 // object.
-func GetCredentialsFromSecretRef(ctx context.Context, client client.Client, secretRef corev1.SecretReference) (*Credentials, error) {
+func GetCredentialsFromSecretRef(ctx context.Context, client client.Client, secretRef corev1.SecretReference, allowDNSKeys bool) (*Credentials, error) {
```

Comment:
Why "allow"DNSKeys?

Reply:
Because they can be either allowed or not. If they are allowed, then a secret containing them will be parsed successfully. If not, parsing such a secret will fail. The idea is, that a secret that contains DNS keys but is not used for DNS will still fail to be parsed (we don't want to suddenly allow a new set of secret keys for everything).

Reply:
Makes sense, thank you!